var/spack/repos/builtin/packages/apex/package.py

```diff
@@ -4,30 +4,141 @@
+        args = []
         spec = self.spec
-        return [
-            '-DBOOST_ROOT=%s' % spec['boost'].prefix,
-            '-DUSE_BFD=TRUE',
-            '-DBFD_ROOT=%s' % spec['binutils'].prefix,
-            '-DUSE_ACTIVEHARMONY=TRUE',
-            '-DACTIVEHARMONY_ROOT=%s' % spec['activeharmony'].prefix,
-            '-DUSE_OMPT=TRUE',
-            '-DOMPT_ROOT=%s' % spec['ompt-openmp'].prefix,
-        ]
+        # CMake variables were updated in version 2.3.0, to make
+        prefix = 'APEX_WITH'
+        test_prefix = 'APEX_'
+        if '@2.2.0' in spec:
+            prefix = 'USE'
+            test_prefix = ''
+
+        if '+cuda' in spec:
+            args.append('-DAPEX_WITH_CUDA=TRUE')
+        else:
+            args.append('-DAPEX_WITH_CUDA=FALSE')
+
+        if '+binutils' in spec:
+            args.append('-DBFD_ROOT={0}'.format(spec['binutils'].prefix))
+            args.append('-D' + prefix + '_BFD=TRUE')
+        else:
+            args.append('-D' + prefix + '_BFD=FALSE')
+
+        if '+activeharmony' in spec:
+            args.append('-DACTIVEHARMONY_ROOT={0}'.format(
+                spec['activeharmony'].prefix))
+            args.append('-D' + prefix + '_ACTIVEHARMONY=TRUE')
+        else:
+            args.append('-D' + prefix + '_ACTIVEHARMONY=FALSE')
+
+        if '+plugins' in spec:
+            args.append('-D' + prefix + '_PLUGINS=TRUE')
+        else:
+            args.append('-D' + prefix + '_PLUGINS=FALSE')
+
+        if '+lmsensors' in spec:
+            args.append('-D' + prefix + '_LM_SENSORS=TRUE')
+        else:
+            args.append('-D' + prefix + '_LM_SENSORS=FALSE')
+
+        if '+mpi' in spec:
+            args.append('-D' + prefix + '_MPI=TRUE')
+        else:
+            args.append('-D' + prefix + '_MPI=FALSE')
+
+        if '+otf2' in spec:
+            args.append('-DOTF2_ROOT={0}'.format(spec['otf2'].prefix))
+            args.append('-D' + prefix + '_OTF2=TRUE')
+        else:
+            args.append('-D' + prefix + '_OTF2=FALSE')
+
+        if '+openmp' in spec:
+            args.append('-D' + prefix + '_OMPT=TRUE')
+        else:
+            args.append('-D' + prefix + '_OMPT=FALSE')
+
+        if '+gperftools' in spec:
+            args.append('-DGPERFTOOLS_ROOT={0}'.format(
+                spec['gperftools'].prefix))
+            args.append('-D' + prefix + '_TCMALLOC=TRUE')
+        else:
+            args.append('-D' + prefix + '_TCMALLOC=FALSE')
+
+        if '+jemalloc' in spec:
+            args.append('-DJEMALLOC_ROOT={0}'.format(spec['jemalloc'].prefix))
+            args.append('-D' + prefix + '_JEMALLOC=TRUE')
+        else:
+            args.append('-D' + prefix + '_JEMALLOC=FALSE')
+
+        if '+boost' in spec:
+            args.append('-DBOOST_ROOT={0}'.format(spec['boost'].prefix))
+
+        if '+tests' in spec:
+            args.append('-D' + test_prefix + 'BUILD_TESTS=TRUE')
+        else:
+            args.append('-D' + test_prefix + 'BUILD_TESTS=FALSE')
```

Comment:
Here you can do:
```suggestion
        args.append(self.define_from_variant('APEX_BUILD_TESTS', 'tests'))
```
and similarly for all the other ones. If it makes sense you can construct a list literal that way:
```python
args = [
    self.define_from_variant(...),
    self.define_from_variant(...),
    self.define_from_variant(...)
]
```

Reply:
done for all cmake variables, thanks